### PR TITLE
Update `BTPayPalRequest`, `BTPayPalVaultRequest` and `BTPayPalCheckoutRequest`

### PR DESCRIPTION
**Pull request by agedd**
### Summary of changes

- Update `BTPayPalRequest`, `BTPayPalVaultRequest`,  and `BTPayPalCheckoutRequest`

### Checklist

- [x] Added a changelog entry

- @agedd

### CODE REVIEW

File: Sources/BraintreePayPal/BTPayPalCheckoutRequest.swift [hunk 1]
```diff
@@ -58,38 +58,23 @@ import BraintreeCore
...
+    var userPhoneNumber: BTPayPalPhoneNumber?
+    var lineItems: [BTPayPalLineItem]?
 
     // MARK: - Initializer
 
     /// Initializes a PayPal Native Checkout request
     /// - Parameters:
-    ///   - amount: Used for a one-time payment. Amount must be greater than or equal to zero, may optionally contain exactly 2 decimal places separated by '.'
+    ///   - amount: Used for a one-time payment. Amount must be greater than or equal to zero, may optionally contain exactly 2 decimal places separated by '.' and is limited to 7 digits before the decimal point.
     ///   - intent: Optional: Payment intent. Defaults to `.authorize`. Only applies to PayPal Checkout.
```
Comment (jaxdesmarais) [inline Sources/BraintreePayPal/BTPayPalCheckoutRequest.swift:78]: If this only applies to the checkout flow what do we think about moving it out of here and only requiring it on the checkout request? We should validate this docstring is true, but worth looking into.

File: Sources/BraintreePayPal/BTPayPalCheckoutRequest.swift [hunk 2]
```diff
@@ -98,21 +83,30 @@ import BraintreeCore
...
     ///   - requestBillingAgreement: Optional: If set to `true`, this enables the Checkout with Vault flow, where the customer will be prompted to consent to a billing agreement
     ///   during checkout. Defaults to `false`.
+    ///   - userAuthenticationEmail: Optional: User email to initiate a quicker authentication flow in cases where the user has a PayPal Account with the same email.
+    ///   - userPhoneNumber: Optional: A user's phone number to initiate a quicker authentication flow in the scenario where the user has a PayPal account
+    /// identified with the same phone number.
+    ///   - lineItems: Optional: The line items for this transaction. It can include up to 249 line items.
     public init(
-        amount: String,
-        intent: BTPayPalRequestIntent = .authorize,
+        amount: String?,
```
Comment (jaxdesmarais) [inline Sources/BraintreePayPal/BTPayPalCheckoutRequest.swift:91]: It looks like amount should be required but we should verify

File: Sources/BraintreePayPal/BTPayPalCheckoutRequest.swift [hunk 3]
```diff
@@ -98,21 +83,30 @@ import BraintreeCore
...
+        intent: BTPayPalRequestIntent? = .authorize,
```
Comment (jaxdesmarais) [inline Sources/BraintreePayPal/BTPayPalCheckoutRequest.swift:92]: General question, have we checked the Obj-C interface for this? Just want to verify the optional enum is okay there.

File: CHANGELOG.md [hunk 4]
```diff
@@ -15,6 +15,8 @@
     * Update `BTLocalPaymentRequest` to make all properties accessible on the initializer only vs via the dot syntax.
   * BraintreeCard 
     * Remove `BTCardRequest`, use `BTCard` directly instead
+  * BraintreePayPal
+    * Update `BTPayPalRequest`, `BTPayPalVaultRequest` and `BTPayPalCheckoutRequest` to make all properties accessible on the initializer only vs via the dot syntax.
```
Comment (stechiu) [inline CHANGELOG.md:19]: Looks like typo "...only use via"

File: Sources/BraintreePayPal/BTPayPalCheckoutRequest.swift (hunk 1 above)
Comment (stechiu) [reply Sources/BraintreePayPal/BTPayPalCheckoutRequest.swift:78]: Side note, I'm surprised we allow transaction that's equal to $0.00

File: UnitTests/BraintreePayPalTests/BTPayPalRequest_Tests.swift [hunk 5]
```diff
@@ -46,12 +46,6 @@ class BTPayPalRequest_Tests: XCTestCase {
         request.riskCorrelationID = "123-correlation-id"
         request.merchantAccountID = "merchant-account-id"
         request.isShippingAddressEditable = true
-        
-        let lineItem = BTPayPalLineItem(quantity: "1", unitAmount: "1", name: "item", kind: .credit)
```
Comment (jaxdesmarais) [inline UnitTests/BraintreePayPalTests/BTPayPalRequest_Tests.swift:50]: I think we want to keep these  in the unit test to be sure we are passing line items as expected.

File: Demo/Application/Features/PayPalWebCheckoutViewController.swift [hunk 6]
```diff
@@ -135,21 +135,22 @@ class PayPalWebCheckoutViewController: PaymentButtonBaseViewController {
         sender.setTitle("Processing...", for: .disabled)
         sender.isEnabled = false
 
-        let request = BTPayPalCheckoutRequest(amount: "5.00")
-        request.userAuthenticationEmail = emailTextField.text
-        request.userPhoneNumber = BTPayPalPhoneNumber(
-            countryCode: countryCodeTextField.text ?? "",
-            nationalNumber: nationalNumberTextField.text ?? ""
+        let request = BTPayPalCheckoutRequest(
```
Comment (jaxdesmarais) [inline Demo/Application/Features/PayPalWebCheckoutViewController.swift:138]: Lets add back in the line items as well

File: Sources/BraintreePayPal/BTPayPalRequest.swift [hunk 7]
```diff
@@ -115,10 +99,8 @@ import BraintreeCore
         landingPageType: BTPayPalRequestLandingPageType = .none,
         displayName: String? = nil,
         merchantAccountID: String? = nil,
-        lineItems: [BTPayPalLineItem]? = nil,
```
Comment (jaxdesmarais) [inline Sources/BraintreePayPal/BTPayPalRequest.swift:118]: I think this is available for both checkout and vault. We can keep line items here and pass it through on the super inits.

File: Sources/BraintreePayPal/BTPayPalCheckoutRequest.swift [hunk 8]
```diff
@@ -58,38 +58,20 @@ import BraintreeCore
...
+    var requestBillingAgreement: Bool
 
     // MARK: - Initializer
 
     /// Initializes a PayPal Native Checkout request
     /// - Parameters:
-    ///   - amount: Used for a one-time payment. Amount must be greater than or equal to zero, may optionally contain exactly 2 decimal places separated by '.'
+    ///   - amount: Used for a one-time payment. Amount must be greater than or equal to zero, may optionally contain exactly 2 decimal places separated by '.' and is limited to 7 digits before the decimal point.
     ///   - intent: Optional: Payment intent. Defaults to `.authorize`. Only applies to PayPal Checkout.
     ///   and is limited to 7 digits before the decimal point.
```
Comment (richherrera) [inline Sources/BraintreePayPal/BTPayPalCheckoutRequest.swift:76]: is this part of the `intent` docstring? do we need this comment here?

Comment (agedd) [reply Sources/BraintreePayPal/BTPayPalCheckoutRequest.swift]: ahh good catch! resolved in 3a8a542bde556b5aa93572f51a7a1a213e24fe1d

File: Sources/BraintreePayPal/BTPayPalRequest.swift [hunk 9]
```diff
@@ -52,59 +52,49 @@ import BraintreeCore
...
     
     // MARK: - Static Properties
     
     static let callbackURLHostAndPath: String = "onetouch/v1/"
 
     // MARK: - Initializer
 
+    /// Initialize a new `BTPayPalRequest`
+    /// - Parameters:
+    ///    - hermesPath: Required :nodoc: Exposed publicly for use by PayPal Native Checkout module. This property is not covered by semantic versioning.
```
[the remark this answers is not in the document]
Comment (saperi22) [reply Sources/BraintreePayPal/BTPayPalRequest.swift:80]: should we be removing it instead?

File: Sources/BraintreePayPal/BTPayPalRequest.swift [hunk 10]
```diff
@@ -52,59 +52,49 @@ import BraintreeCore
...
+    ///    - paymentType: Required: The payment type, either checkout or vault :nodoc: Exposed publicly for use by PayPal Native Checkout module. This property is not covered by semantic versioning.
+    ///    - isShippingAddressRequired: Defaults to false. When set to true, the shipping address selector will be displayed.
+    ///    - isShippingAddressEditable: Defaults to false. Set to true to enable user editing of the shipping address. Only applies when `shippingAddressOverride` is set.
+    ///    - localeCode: Optional: A locale code to use for the transaction.
+    ///    - shippingAddressOverride: Optional: A valid shipping address to be displayed in the transaction flow. An error will occur if this address is not valid.
+    ///    - landingPageType: Optional: Landing page type. Defaults to `.none`.  
+    ///     - Note: Setting the BTPayPalRequest's landingPageType changes the PayPal page to display when a user lands on the PayPal site to complete the payment.
+    ///        `.login` specifies a PayPal account login page is used.
+    ///        `.billing` specifies a non-PayPal account landing page is used.
```
[the remark this answers is not in the document]
Comment (agedd) [reply Sources/BraintreePayPal/BTPayPalRequest.swift:89]: hmm i see, not sure why it isn't showing up for you - i'm able to see it when i click on `Show Quick Help` on the `init` as here: 

Screenshot 2024-11-26 at 3 01 35 PM

**Review by richherrera — APPROVED** [review]

**Pull request merged**